src/pytorch/CFConvNeighbors.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2020-2021 Acellera
+ * Authors: Raimondas Galvelis
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include "CFConvNeighbors.h"
+#include "CpuCFConv.h"
+
+#ifdef ENABLE_CUDA
+#include <stdexcept>
+#include <cuda_runtime.h>
+// #include <c10/cuda/CUDAStream.h>
+#include "CudaCFConv.h"
+
+#define int2 int[2]
+#define float3 float[3]
+
+#define CHECK_CUDA_RESULT(result) \
+    if (result != cudaSuccess) { \
+        throw std::runtime_error(std::string("Encountered error ")+cudaGetErrorName(result)+" at "+__FILE__+":"+std::to_string(__LINE__));\
+    }
+#endif
+
+namespace NNPOps {
+namespace CFConvNeighbors {
+
+using HolderPtr = torch::intrusive_ptr<Holder>;
+
+Holder::Holder(double cutoff) : torch::CustomClassHolder(), cutoff(cutoff), numAtoms(0), device(torch::kCPU), impl() {}
```

Comment:
This should likewise be `impl(nullptr)`.

Reply:
std::shared_prt() and std::shared_ptr(nullptr) are equivalent (https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr).

Reply:
OK! Initializing explicitly (https://github.com/openmm/NNPOps/pull/40/commits/15711e2d91fa550fcfac028323d693a109f87cf1)